### DISCUSSION TITLE
OpenAI assistant InvokeStreamingAsync - chat history

### DISCUSSION THREAD

**Discussion by JP-droidic** [microsoft/semantic-kernel#8810] [Q&A] [answered]
Hi team

Why is chat history mandatory when we invoke OpenAi Assistant.InvokeStreamingAsync - the first parameter is threadId - why do we need to keep chat history?

**Comment by evchaki** [comment]
@crickman - can you take a look at this for .NET?

**Comment by crickman** [comment, marked as answer]
Hi @JP-droidic, thank you for calling this out.

Two scenarios require maintaining the _non-streamed_ history: `ChatCompletionAgent` and `AgentChat`.  The `AgentChat` case extends to both `ChatCompletionAgent` and `OpenAIAssistantAgent`.  This focus, resulted in the presence of the history parameter to which  you are referring.

Now that you've called it out, I realize that for this _no-chat_ invocation case...its only a convenience and not required.  I can certainly make it optional.  

I like have the option available as a convenience since we know some customers cases involve examining the history...even for streaming.  Forcing the caller to build history from streaming content can be tedious so maintaining this as an option can be helpful for these cases.

**Reply by JP-droidic** [reply]
This makes sense. Thank you!

**Comment by crickman** [comment]
https://github.com/microsoft/semantic-kernel/issues/8857
